client/java/src/main/java/io/openlineage/client/utils/gravitino/SparkGravitinoInfoProvider.java

```diff
@@ -0,0 +1,205 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.utils.gravitino;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Provider for Gravitino configuration from Spark session. Reads Gravitino-specific configuration
+ * from Spark's runtime configuration using reflection to avoid direct Spark dependencies.
+ */
+@Slf4j
+public class SparkGravitinoInfoProvider implements GravitinoInfoProvider {
+
+  private static final String SPARK_SESSION_CLASS_NAME = "org.apache.spark.sql.SparkSession";
+  private static final String SPARK_RUN_CONFIG_CLASS_NAME = "org.apache.spark.sql.RuntimeConfig";
+
+  /** Configuration key for Gravitino metalake name used by GVFS filesystem */
+  public static final String metalakeConfigKeyForFS = "spark.hadoop.fs.gravitino.client.metalake";
+
+  /** Configuration key for Gravitino metalake name used by Gravitino Spark connector */
+  public static final String metalakeConfigKeyForConnector = "spark.sql.gravitino.metalake";
+
+  /**
+   * Configuration key to control whether to use Gravitino identifier format. Default: true when not
+   * set
+   */
+  public static final String useGravitinoConfigKey = "spark.sql.gravitino.useGravitinoIdentifier";
+
+  /**
+   * Configuration key for catalog name mappings. Format:
+   * "spark_catalog1:gravitino_catalog1,spark_catalog2:gravitino_catalog2"
+   */
+  public static final String catalogMappingConfigKey = "spark.sql.gravitino.catalogMappings";
+
+  private static final int CATALOG_MAPPING_PARTS = 2;
+
+  @Override
+  public boolean isAvailable() {
+    try {
+      SparkGravitinoInfoProvider.class.getClassLoader().loadClass(SPARK_SESSION_CLASS_NAME);
+      log.debug("SparkSession class found, Gravitino info provider is available");
```

Comment:
I don't understand this - why would `SparkSession` existence mean that Gravitino is available?